### PR TITLE
Update Loadable Plug-in section

### PR DESCRIPTION
**Pull request by mhmohona**
## Explanation
Update Loadable Plug-in section - expanded explanation and added flowchart for better understanding of the workflow. 

## What type of PR is this
/kind documentation

### CODE REVIEW

**Comment by alabulei1** [comment]
Hello, I am a [code review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

**Review by alabulei1 — APPROVED** [review]

**Comment by alabulei1** [comment]
@q82419 Could you please helo review this?

**Comment by adithyaakrishna** [comment]
@mhmohona Could you please fix the DCO check?

File: docs/contribute/plugin/intro.md [hunk 1]
```diff
@@ -14,9 +14,40 @@ In current, developers can follow the guides to implement the plug-ins in [C API
...
+Loadable plugins are standalone shared libraries (`.so`/`.dylib`/`.dll` files) that can be loaded by the WasmEdge runtime environment at runtime. These plugins can provide additional functionality to the WasmEdge runtime environment, such as new modules that can be imported by WebAssembly modules.
 
-Please [refer to the plugin example code](https://github.com/WasmEdge/WasmEdge/tree/master/examples/plugin/get-string).
+To create a loadable plugin for WasmEdge, developers can use the WasmEdge Plugin SDK, which provides a set of C/C++ APIs for creating and registering plugins. The SDK also includes [example code](https://github.com/WasmEdge/WasmEdge/tree/master/examples/plugin/get-string) that demonstrates how to create a simple plugin that returns a string.
+
+Once a loadable plugin has been created, it can be loaded by the WasmEdge runtime environment using the WasmEdge_LoadWasmFromFile or WasmEdge_LoadWasmFromBuffer API. The plugin can then be used to provide functionality to WebAssembly modules, such as access to system resources or specialized hardware.
```
Comment (q82419) [inline docs/contribute/plugin/intro.md:21]: The API names should use `` quotes.

**Review by q82419 — COMMENTED** [review]
You lost the most important part.
A plug-in can be loaded only after loading that plug-in from file with APIs.
Refer to the steps:
https://wasmedge.org/book/en/sdk/c/ref.html#plug-ins

File: docs/contribute/plugin/intro.md (hunk 1 above)
Comment (mhmohona) [reply docs/contribute/plugin/intro.md:21]: oh, thanks for pointing. I am adding it.

File: docs/contribute/plugin/intro.md [hunk 2]
```diff
@@ -14,9 +14,40 @@ In current, developers can follow the guides to implement the plug-ins in [C API
 
 ## Loadable Plug-in
 
-Loadable plugin is a standalone `.so`/`.dylib`/`.dll` file that WasmEdge can load during runtime environment, and provide modules to be imported.
+Loadable plugins are standalone shared libraries (`.so`/`.dylib`/`.dll` files) that can be loaded by the WasmEdge runtime environment at runtime. These plugins can provide additional functionality to the WasmEdge runtime environment, such as new modules that can be imported by WebAssembly modules.
 
-Please [refer to the plugin example code](https://github.com/WasmEdge/WasmEdge/tree/master/examples/plugin/get-string).
+To create a loadable plugin for WasmEdge, developers can use the WasmEdge Plugin SDK, which provides a set of C/C++ APIs for creating and registering plugins. The SDK also includes [example code](https://github.com/WasmEdge/WasmEdge/tree/master/examples/plugin/get-string) that demonstrates how to create a simple plugin that returns a string.
```
Comment (adithyaakrishna) [inline docs/contribute/plugin/intro.md:19]:
```suggestion
To create a loadable plugin for WasmEdge, developers can use the WasmEdge Plugin SDK, which provides a set of Rust or C/C++ APIs for creating and registering plugins. The SDK also includes [example code](https://github.com/WasmEdge/WasmEdge/tree/master/examples/plugin/get-string) that demonstrates how to create a simple plugin that returns a string.
```

File: docs/contribute/plugin/intro.md [hunk 3]
```diff
@@ -14,9 +14,40 @@ In current, developers can follow the guides to implement the plug-ins in [C API
...
+    F --> G
+  </div>
+</div>
+
+This flowchart shows the basic steps involved in creating and using a loadable plugin in WasmEdge. The first step is to create the plugin, which involves writing the code for the plugin functions and compiling the code into a shared library.
+
+Once the plugin has been compiled, it can be loaded by the WasmEdge runtime environment using the `WasmEdge_LoadWasmFromFile` or `WasmEdge_LoadWasmFromBuffer` API. The plugin can then be registered with the runtime environment using the `WasmEdge_RegisterImport` API.
+
+After the plugin has been registered, a WebAssembly module can be loaded using the `WasmEdge_LoadWasmFromFile` or `WasmEdge_LoadWasmFromBuffer` API. The module can then import the plugin module using the `WasmEdge_VMRegisterModule` API.
+Finally, the WebAssembly module can execute its functions, which may call the functions provided by the plugin module. The plugin functions can then perform their specific tasks, such as accessing system resources or specialized hardware.
```
Comment (adithyaakrishna) [inline docs/contribute/plugin/intro.md:50]:
```suggestion
After the plugin has been registered, a WebAssembly module can be loaded using the `WasmEdge_LoadWasmFromFile` or `WasmEdge_LoadWasmFromBuffer` API. The module can then import the plugin module using the `WasmEdge_VMRegisterModule` API.

Finally, the WebAssembly module can execute its functions, which may call the functions provided by the plugin module. The plugin functions can then perform their specific tasks, such as accessing system resources or specialized hardware.
```

**Comment by mhmohona** [comment]
> You lost the most important part. A plug-in can be loaded only after loading that plug-in from file with APIs. Refer to the steps: https://wasmedge.org/book/en/sdk/c/ref.html#plug-ins

Hello @q82419, shall I add these [steps](https://wasmedge.org/book/en/sdk/c/ref.html#plug-ins) in the documents? If yes, shall I create a new page for Loadable Plug-in and add there? As if I add the steps here the page will be too long and reader might lose focus.

**Comment by q82419** [comment]
> > You lost the most important part. A plug-in can be loaded only after loading that plug-in from file with APIs. Refer to the steps: https://wasmedge.org/book/en/sdk/c/ref.html#plug-ins
> 
> Hello @q82419, thank you for your kind feedback. Regarding updating the section I have one confusion - shall I add these [steps](https://wasmedge.org/book/en/sdk/c/ref.html#plug-ins) in the documents for plugin guideline?

You should add these steps.
After loading the plugins (by all default paths or a given plugin path), you can then get the plugin context to create the module instance.

**Comment by mhmohona** [comment]
Reopened another PR to pass DOC test - https://github.com/WasmEdge/docs/pull/106

**Pull request closed without merging**